Fixed Bug With Executing Git From Path With Spaces

MarvelProgramming opened the pull request:
-Added quotes around the path that the "exec" function uses when searching for the Git executable

For issue #33

Commit:
Fixed Bug With Executing Git From Path With Spaces

-Added quotes around the path that the "exec" function uses when searching for the Git executable

rmkubik commented on scripts/objectiveValidation.js:
```diff
@@ -10,7 +10,7 @@ async function execGitWithArgs(helper, argString, execOptions = {}) {
     );
   }
 
-  return exec(`${TQ_GIT_EXE} ${argString}`, execOptions);
```
Let's add a check before this for leading and trailing `"` and `'` in the TQ_GIT_EXE env var contents.

We should throw a new error if these exist with a message:
> TwilioQuest no longer requires you to surround your git executable path with quotation marks if it contains spaces.

Let's also put a comment to say that we advised users to add quotations marks to get past this validator after the TQ 3.10.0 release introduced the bug.

Commit:
Updated objectiveValidation

-Added check for single and double quotes in TQ_GIT_EXE, which throws a new exception. Users were directed to add quotes to their paths until a
fix was deployed. This ensures these users know how to correct their variable once this goes live
-Added comment

alvinphilips commented:
Hi, I was just going through this myself. Would escaping the spaces (by prepending the spaces with `\\`) instead be a cleaner approach? I think it may also eliminate the need to throw an Error if a user does include the quotation marks.

MarvelProgramming commented:
> Hi, I was just going through this myself. Would escaping the spaces (by prepending the spaces with `\\`) instead be a cleaner approach? I think it may also eliminate the need to throw an Error if a user does include the quotation marks.

Correct me if I'm wrong, but I don't think this works on Windows? I tested it in PowerShell and got an `InvalidArgument` error

Commit:
Change && to ||

I have changed the && to || in the parenthetical comparison. The && check does make sense and checks for the specific situation we were talking about. However, doing a || allows us to catch other situations where a user has entered an incorrect path without hurting the original test case.
